lib/web_ui/lib/src/engine/canvaskit/picture.dart

```diff
@@ -15,17 +15,18 @@ import 'surface_factory.dart';
 
 /// Implements [ui.Picture] on top of [SkPicture].
 class CkPicture implements ui.Picture {
```

Comment:
I actually made this abstract class that you might want to conform to here: https://github.com/flutter/engine/blame/893ab3bf7bb9cf104846ec50b79baf4864089253/lib/web_ui/lib/src/engine/scene_painting.dart#L15C3-L15C3

I actually wrote Skwasm's scene builder in an agnostic way where it could theoretically be reused for any renderer that implements the normal `dart:ui` interfaces plus a few odds and ends that aren't exposed (such as this `cullRect` function). It might be interesting to see if CanvasKit could be migrated to that scene builder and do some perf measurements at some point, I think it does a little less work in some cases than the CanvasKit one (although it doesn't support platform views yet, so we wouldn't want to switch over to that until the platform views are implemented).

I also think a getter for `cullRect` makes more sense than a function, but up to you.

Reply:
Thanks! I'm getting started on the `filterBounds` CanvasKit change

Reply:
https://skia-review.googlesource.com/c/skia/+/721299 is out for review

Reply:
Done.